src/dataTransformers/NumberDataTransformer.ts

```diff
@@ -0,0 +1,54 @@
+import { InvalidParameterError, TransformDataError } from '../errors';
+import type { DataTransformer } from '../types';
+
+/**
+ * Number Data Transformer
+ * @throws {@link InvalidParameterError} An invalid parameter was passed to the function
+ */
+class NumberDataTransformer implements DataTransformer {
+	/** Number of implied decimals in database storage */
+	private readonly dbDecimals: number;
+
+	public constructor(dbDecimals = 0) {
+		if (!Number.isInteger(dbDecimals)) {
```

Comment:
This should also validate that dbDecimals is a positive interger

Reply:
Good call.  This wasn't in the original schema type but it makes sense to add.